src/esp/sim/Simulator.h

```diff
@@ -112,7 +112,12 @@ class Simulator {
    * simulator.  This will only be nonzero if the simulator is built in
    * --headless mode on linux
    */
-  int gpuDevice() const { return context_->gpuDevice(); }
+  int gpuDevice() const {
+    if (context_ == nullptr) {
+      return 0;
```

Comment:
Is device 0 the default GPU device in UI mode?
Then if context_ is NULL, does it mean the context is not built (which means it is illegal to call this function), and we should return -1 maybe?

In my opinion:
Do we need to just quit and prompt an error (using CORRADE_ASSERT)? It does not make any sense to tolerate this error.

Reply:
+1 for the assert :)

Reply:
AFAIR, there is some issue with the order in which these things get called. gpuDevice() is probably getting called before the context is built? I remember my python viewer crashing without this null check.

Reply:
@dexter1691 : If you read the code (next line) here, `context_->gpuDevice()`, how could `gpuDevice()` get called before the context is constructed?